Rx/v2/src/rxcpp/operators/rx-reduce.hpp

```diff
@@ -230,7 +256,35 @@ struct average {
             }
             return avg;
         }
-        return a.value;
+        throw rxcpp::empty_error("No elements");
+    }
+};
+
+template<class T>
+struct sum {
+    struct seed_type
+    {
+        seed_type()
+            : value()
+            , has_accumulation(false)
+        {
+        }
+        T value;
+        bool has_accumulation;
+    };
+    seed_type seed() {
+        return seed_type{};
+    }
+    seed_type operator()(seed_type a, T v) {
+        a.value += v;
```

Comment:
Accumulate should just reset to the initial value (when the seed is empty) and only add when the seed is !empty. Also, use + not +=. sum should work for any type that has defined operator+() even if they have not defined operator+=().